### PR TITLE
Develop: Top Three Refactor

### PR DESCRIPTION
**Pull request by Tthomas63**
Refactor Top-Three to use the same data/format as results.

### CODE REVIEW

**Comment by Tthomas63** [comment]
Problems:
 - Alignment with the rest of the page
 - Some competitions do not display results or top three (?)
 - Add back the `no data` condition

File: codalab/apps/web/templates/web/competitions/view.html [hunk 1]
```diff
@@ -229,52 +229,88 @@ <h3 class="panel-title"><strong>Competition Ends</strong></h3>
...
+        <div class="col-md-4 well">
+            {% for group in top_three %}
+                <table class="table">
+                    <tr class="">
+                        <th rowspan="2">#</th>
+                        <th rowspan="2">User</th>
+                        {% if phase.competition.allow_teams %}
+                            <th rowspan="2">Team Name</th>
+                        {% endif %}
+                        {% for head in group.headers|slice:"0:1"%}
```
Comment (ckcollab) [inline codalab/apps/web/templates/web/competitions/view.html]: Could you do these operations in Python and just pass the same thing as before to top 3?

File: codalab/apps/web/views.py [hunk 2]
```diff
@@ -536,11 +535,20 @@ def get_context_data(self, **kwargs):
                         'counts': [s['count'] for s in qs],
                         'sorting': score_def.sorting,
                     }
-
-                context['top_three_leaders'] = self.get_object().get_top_three()
+                # Below is where we refactored top_three context.
+                context['top_three'] = context['active_phase'].scores()
```
Comment (ckcollab) [inline codalab/apps/web/views.py]: we can keep this behavior in `get_top_three()`? did you remove `get_top_three()` if we're not using it any more?

Comment (Tthomas63) [reply codalab/apps/web/views.py]: I left `get_top_three()` for now.

File: codalab/apps/web/views.py [hunk 3]
```diff
@@ -536,11 +535,35 @@ def get_context_data(self, **kwargs):
...
+                        first_score = values[0]['val']
+                        top_three_list.append(
+                            {
+                                "username": scoredata['username'],
+                                "score": first_score
+                            }
+                        )
+
+                        # Regular score stuff
+                        sub = models.CompetitionSubmission.objects.get(pk=scoredata['id'])
```
Comment (ckcollab) [inline codalab/apps/web/views.py]: pull this

**Comment by Tthomas63** [comment]
I pushed this onto the LRI test branch for this. Seems to be working nicely

File: codalab/apps/web/templates/web/competitions/view.html [hunk 4]
```diff
@@ -229,54 +229,29 @@ <h3 class="panel-title"><strong>Competition Ends</strong></h3>
...
+                    <th>#</th>
+                    <th>Username</th>
+                    <th>Score</th>
+                </tr>
+                </thead>
+                <tbody>
+                {% if not active_phase.is_blind %}
+                    {% for leader in top_three %}
+                        <tr>
+                            <td>{{ forloop.counter }}</td>
```
Comment (ckcollab) [inline codalab/apps/web/templates/web/competitions/view.html]: what will this display with empty top 3?

File: codalab/apps/web/views.py [hunk 5]
```diff
@@ -514,9 +514,8 @@ def get_context_data(self, **kwargs):
         context['current_server_time'] = datetime.now()
 
         context["first_phase"], context["previous_phase"], context['active_phase'], context["next_phase"] = get_first_previous_active_and_next_phases(competition)
+        context['phase'] = context['active_phase']
```
Comment (ckcollab) [inline codalab/apps/web/views.py]: still needed?

File: codalab/apps/web/views.py (hunk 2 above)
Comment (ckcollab) [reply codalab/apps/web/views.py]: hmm, think we should pull it, though?

File: codalab/apps/web/templates/web/competitions/view.html (hunk 4 above)
Comment (Tthomas63) [reply codalab/apps/web/templates/web/competitions/view.html]: Blank table

File: codalab/apps/web/views.py (hunk 5 above)
Comment (Tthomas63) [reply codalab/apps/web/views.py]: No should not be, good catch.

File: codalab/apps/web/views.py (hunk 2 above)
Comment (Tthomas63) [reply codalab/apps/web/views.py]: I removed that method.

File: codalab/apps/web/templates/web/competitions/view.html (hunk 4 above)
Comment (ckcollab) [reply codalab/apps/web/templates/web/competitions/view.html]: NOTE: Please hide top 3 if nothing should be shown

**Pull request merged**